### PR TITLE
Add trailing slash to current_url to fix matching URLs without trailing slash

### LINKED ISSUE #1784: URLs without ending slash aren't fully recognized
## Description

In the https://github.com/GlotPress/GlotPress/pull/1701 the base URL of GlotPress seems to be solved, redirecting to the Projects page, either having ending glotpress/ slash or not.

For all the remaining URLs there seems to happen a slight problem.
If the URL is `site.com/glotpress/languages/`, what seems to be a fully qualified URL for GlotPress, the page is loaded correctly.
If the URL is `site.com/glotpress/page/`, a 404 is correctly rendered.
But, if the URL is `site.com/glotpress/languages` without trailing slash, GlotPress finds the Locales page, but in the menu it's not matched as currently in the Locales page, because the bottom border of the menu item isn't highlighted.

Not sure what would be better, to redirect and add the trailing slash when the slug is matched, or try to match the slug with or without ending slash for menu `current` identifying purposes.

This is valid for every page, Locales, Projects, Profile, Settings.

## Step-by-step reproduction instructions

Please list the steps needed to reproduce the bug. For example:
1. Go to `site.com/glotpress/projects/` (**with** ending slash)
2. See the below screenshot
imagem

3. Go to `site.com/glotpress/projects` (**without** ending slash)
4. See the below screenshot
imagem

## Expected behavior
GlotPress should either recognize the URL without ending slash, or redirect.

### PR DESCRIPTION
**Pull request by pedro-mendonca**
## Problem

Navigating to a page like `site.com/glotpress/projects` without the trailing slash won't identify the current page in the top menu.

## Solution

Add trailing slash when trying to match menu items to show the current one.

Fixes #1784

### CODE REVIEW

**Comment by swissspidy** [comment]
This makes sense to me 👍 

Every once in awhile I install GlotPress again and get 404s because I forget the slashes.

**Comment by pedro-mendonca** [comment]
This PR actually doesn't fix that.
In the related issue I mention that, this is currently fixing only the matching of the url with or without slash for the menu item bottom border of the current menu.

**Comment by swissspidy** [comment]
Oh, gotcha!

File: gp-includes/template.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ function gp_nav_menu( $location = 'main' ) {
 	$items = gp_nav_menu_items( $location );
 
 	// Get the current URI.
-	$current_uri = str_replace( home_url(), '', gp_url_current() );
+	$current_uri = str_replace( home_url(), '', trailingslashit( gp_url_current() ) );
```
Comment (amieiro) [inline gp-includes/template.php]: It is a bit weird when you put a URL without the ending slash and the current link in the menu is marked. What do you think about adding a redirect to the URL with the ending slash when the user goes to a URL without this ending slash?

```suggestion
	$current_uri           = str_replace( home_url(), '', trailingslashit( gp_url_current() ) );
	$uri_no_trailing_slash = str_replace( home_url(), '', gp_url_current() );
	if ( $current_uri !== $uri_no_trailing_slash ) {
		wp_redirect( $current_uri );
	}
```

Comment (pedro-mendonca) [reply gp-includes/template.php]: I like the idea, but isn't redirecting only on `gp_nav_menu()` for matching menus with parent items CSS too late?
What if a specific page don't load `gp_nav_menu()`?
I think that the redirect should be immediatly on page load, while checking the current URL agains the known routes.

**Comment by pedro-mendonca** [comment]
As discussed with @amieiro, changed the approach to check for trailing slash in the URL as soon as possible.
If doesn't exist, redirect to trailingslash URL.

@swissspidy:
> Every once in awhile I install GlotPress again and get 404s because I forget the slashes.

> This PR actually doesn't fix that.
> In the related issue I mention that, this is currently fixing only the matching of the url with or without slash for the menu item bottom border of the current menu.

With the current approach change, this PR actually solves the issue of 404s :)

**Review by amieiro — APPROVED** [review]

**Pull request merged**